docs/src/python/duckdb.md

```diff
@@ -19,14 +22,15 @@ data = [
     {"vector": [5.9, 26.5], "item": "bar", "price": 20.0}
 ]
 table = db.create_table("pd_table", data=data)
-arrow_table = table.to_arrow()
 ```
 
-DuckDB can directly query the `pyarrow.Table` object:
+To query the table, first call `to_lance` to convert the table to a "dataset", which is an object that can be queried by DuckDB. Then all you need to do is reference that dataset by the same name in your SQL query.
```

Comment:
What does "dataset" mean here exactly? Just thinking from the perspective of someone who may not know all this interop stuff. Is this a Lance dataset, or an Arrow dataset, or something in between?

Reply:
It's `LanceDataset`, which duck types like a PyArrow `Dataset`. Do you think I should explain that? I felt that was not that important practically speaking.

Reply:
I feel like just saying "dataset" in quotes makes it sound like it's special in some way without us giving any explanation. Maybe we could just say it's a Lance Dataset that's read by DuckDB is able to view natively as an Arrow dataset?

Reply:
We do drop enough hints throughout the docs that Lance is the underlying format, so I think this would still be useful info to have.